doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -309,28 +358,71 @@ Creating hint databases
+have the `Opaque` setting.  We recommend using :cmd:`Hint Constants`,
+:cmd:`Hint Projections` and :cmd:`Hint Variables` immediately after :cmd:`Create HintDb`
+to make these settings explicit.
+
+Use :cmd:`Hint Opaque` and :cmd:`Hint Transparent` to control the opacity of
+individual items.  Hint opacity settings influence which hints the search tactics
+try, but they may or may not affect how the selected tactic is executed.
+(Usually not, except for :tacn:`typeclasses eauto`, which uses the equivalent of
+:tacn:`autoapply`, which explicitly uses the hint opacity settings when
+applying a :cmd:`Hint Resolve` hint.)
+The proof search tactics use unification to choose which tactics to try, for example
+whether the goal unifies with a theorem given by :cmd:`Hint Resolve`.
+These settings are distinct from the non-hint :cmd:`Opaque` and :cmd:`Transparent`
+settings.
 
 .. cmd:: Create HintDb @ident {? discriminated }
 
-   If there is no hint database named :n:`@name`, creates a new hint database
-   with that name.  Otherwise, does nothing.  The database is
-   implemented by a Discrimination Tree (DT) that serves as a filter to select
-   the lemmas that will be applied. When discriminated, the DT uses
-   transparency information to decide if a constant should considered rigid for
-   filtering, making the retrieval more efficient. By contrast, undiscriminated
-   databases treat all constants as transparent, resulting in a larger
-   number of selected lemmas to be applied, and thus putting more pressure on
-   unification.
+   If there is no hint database named :n:`@ident`, creates a new hint database
+   with that name.  Otherwise, does nothing.
+
+   All constants, variables and projections are set to default to unfoldable (use
+   :cmd:`Hint Constants`, :cmd:`Hint Projections` and :cmd:`Hint Variables`
+   to change this).
+
+   By default, hint databases are undiscriminated.  We recommend using `discriminated`
+   because it generally performs better.
+
+.. _hint_performance_considerations:
+
+Performance considerations
+``````````````````````````
+
+The proof search tactics decide which hints to try by first selecting hints
+whose pattern has the same :term:`head constant` as the goal.  Then the selected
+hints are tried as follows:
+
+- For hints other than :cmd:`Hint Extern`: If the database is discriminated, only
+  matching hints are tried.  Hints match if all compared items that are opaque in
+  both the goal and the pattern are identical.  In this case you can reduce the
+  number of matches by marking definitions as hint opaque through trial and error
+  (if the proof search fails after making something opaque, then it's incorrect).
+
+  Otherwise (if not discriminated), all hints will be tried.  (Therefore, use
+  `discriminated` databases whenever possible.)
+- For `Extern` hints: If the hint has a pattern, hints matching the goal will be tried.
+  In this case, matching uses `syntactic unification
```

Comment:
This information is no longer in this paragraph 
> The database is
   implemented by a Discrimination Tree (DT) that serves as a filter to select
   the lemmas that will be applied.
   
What does discriminate means in this context.